src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -1183,6 +1180,10 @@ int wm_vuldet_send_agent_report(sqlite3 *db, OSHash *cve_table, agent_software *
             }
             w_strdup(pkg->arch, report->arch);
 
+            // Build title
+            os_calloc(OS_SIZE_512, sizeof(char), report->title);
+            snprintf(report->title, OS_SIZE_512, "%s affects %s", report->cve, report->software);
```

Comment:
Hi, the allowed maximum size is the same as the `report->title` buffer size. In a worst case scenario, the last byte would be overwritten, causing invalid reads and even segfaults. 
**Proposed solution:**
` snprintf(report->title, OS_SIZE_512 - 1, "%s affects %s", report->cve, report->software);`